src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -133,6 +133,9 @@ export function generateGraphFromYaml(inputYaml: string): GraphObject | undefine
 
         // If the input yaml does not have a 'Resources' key, no graph can be generated
         if (!_.isObjectLike(resources)) {
+            getLogger().error(
+                'SAM Visualize: Error rendering SAM template. Cannot render a template with no `Resources` key.'
```

Comment:
Is this a message worth surfacing to the user via a toast as well?

Reply:
I think that's a good idea, and I've implemented it. It also helps to add more information:

1. Which resource is not defined correctly / which resource is missing a valid `Type`
2. Including a link to [template anatomy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html)
3. Including YAML parse errors (not sure about this, they're not too clear in a toast.

However, the problem with using `vscode.window.showErrorMessage` is that [you can't make them timeout](https://stackoverflow.com/a/38096947). So these error messages will float around even after the user may have fixed them.

I'll experiment with the `setStatusBarMessage`

Reply:
![image](https://user-images.githubusercontent.com/45087499/130511515-48a34f1f-b04c-4f23-871d-5e64b6a7db52.png)

The setStatusBarMessages are very small, and I'd say most customers will not notice them. You can set these to timeout, but ideally we'd want them to go away after the problem is solved, which is harder.

Reply:
Update: Decided that general error message with link to view logs in console is ideal.